tests/source/issue-3805.rs

```diff
@@ -0,0 +1,43 @@
+// rustfmt-format_macro_matchers: true
+
+// From original issue example - Line length 101
+macro_rules! test {
+($aasdfghj:expr, $qwertyuiop:expr, $zxcvbnmasdfghjkl:expr, $aeiouaeiouaeio:expr, $add:expr) => {{
+return;
+}};
+}
```

Comment:
Can we add a test case where we have nested blocks in the macro body:

```rust
macro_rules! test {
($aasdfghj:expr, $qwertyuiop:expr, $zxcvbnmasdfghjkl:expr, $aeiouaeiouaeio:expr, $add:expr) => {{{{
return;
}}}};
}
```

or even some whitespace between the two opening braces

```rust
macro_rules! test {
($aasdfghj:expr, $qwertyuiop:expr, $zxcvbnmasdfghjkl:expr, $aeiouaeiouaeio:expr, $add:expr) => {    {
return;
}};
}
```

Reply:
Added the test cases.  Note that in the output of first test case below, the 3rd and 4th `{` are starting new indented block, which is different than keeping the 1st and 2nd `{` together.  However I assume this is acceptable.
```rust
macro_rules! test {
    (
        $aasdfghj:expr, $qwertyuiop:expr, $zxcvbnmasdfghjkl:expr, $aeiouaeiouaeio:expr, $add:expr
    ) => {{
        {
            {
                return;
            }
        }
    }};
}
```

Reply:
Yes, that's the intended behavior. the first set of braces denotes the arms of the macro. The second set of braces are part of the code that this macro will generates and will be the outer most block. All other braces should be indented as they define new nested blocks within the macro.